Adapt documentation to new split binaries from C++ rewrite. (#2289)
mendersoftware/mender-docs: 301.Troubleshoot/01.Yocto-project-build/docs.md

    category: docs
---

<!--AUTOVERSION: "Mender clients % have been released"/ignore-->
## Mender clients 4.0.0 have been released but my build still uses the single-client 3.x series

<!--AUTOVERSION: "Yocto branches 4.0 (%) and older"/ignore "Mender clients % and later"/ignore-->

> Mender clients 4.0.0 and later have been split up into several separate components: `mender-auth` and `mender-update`, plus `mender-snapshot` which is now a separate recipe. This also changes the binary names, and just `mender` by itself is not valid anymore. Because of this it is not completely backwards compatible. On the stable Yocto branches 4.0 (kirkstone) and older, the latest major release is the 3.x series, and therefore 4.0 and later are not built by meta-mender unless requested.

Reviewer: ```suggestion
Mender Client version 4.0.0 and later have been split up into several separate components: `mender-auth` and `mender-update`, plus `mender-snapshot` which is now a separate recipe. This also changes the binary names, and just `mender` by itself is not valid anymore. Because of this it is not completely backwards compatible. On the stable Yocto branches 4.0 (kirkstone) and older, the latest major release is the 3.x series, and therefore 4.0 and later are not built by meta-mender unless requested.
```

<!--AUTOVERSION: "To request version % of the clients"/mender-->
To request version 4.0.0 of the clients, add a snippet like the following to `local.conf`:

<!--AUTOVERSION: "PREFERRED_VERSION_mender-client = \"%\""/mender-->
```
PREFERRED_PROVIDER_mender-native =  "mender-native"
PREFERRED_RPROVIDER_mender-auth =   "mender"
PREFERRED_RPROVIDER_mender-update = "mender"
```

<!--AUTOVERSION: "Yocto branches higher than 4.0 (%)"/ignore "Mender client % and later"/ignore-->

> In Yocto branches higher than 4.0 (kirkstone), Mender client 4.0.0 and later will be built by default.

Reviewer: ```suggestion
In Yocto branches higher than 4.0 (kirkstone), Mender Client 4.0.0 and later will be built by default.
```

Unlike x86, ARM based boards usually do not implement [the UEFI boot standard](https://en.wikipedia.org/wiki/Unified_Extensible_Firmware_Interface?target=_blank). Therefore, on ARM, U-Boot is used as a first stage bootloader which provides a UEFI loader, and this is then used to boot GRUB using the UEFI boot standard. For a large number of boards out there, this works out of the box.

> !!! Do not confuse the usage of "U-Boot as a first stage bootloader" with "U-Boot integration". "GRUB integration" still means that GRUB is used for all integration with the Mender client and the rootfs-image update module.

Reviewer: ```suggestion
!!! Do not confuse the usage of "U-Boot as a first stage bootloader" with "U-Boot integration". "GRUB integration" still means that GRUB is used for all integration with the Mender Client and the rootfs-image Update Module.
```
